include/seqan/arg_parse/arg_parse_version_check.h

```diff
@@ -0,0 +1,466 @@
+    {
+        if (!number.empty())
+        {
+            seqan::appendValue(numbers, atoi(seqan::toCString(number)));
+        }
+    }
+}
+
+// ----------------------------------------------------------------------------
+// Function _getVersionNumbers()
+// ----------------------------------------------------------------------------
+inline bool _readVersionNumbers(seqan::String<int> & version_numbers, std::string const & version_file)
+{
+    std::ifstream myfile;
+    myfile.open(version_file.c_str());
+    std::string line;
+    if (myfile.is_open())
+    {
+        std::getline(myfile,line); // get first line which should only contain the version number
+        if (std::regex_match(line, std::regex("^[[:digit:]]\\.[[:digit:]]\\.[[:digit:]]$")))
+        {
+            _getNumbersFromString(version_numbers, line);
+            myfile.close();
+        }
+        else if (line == "UNREGISTERED APP")
+        {
+            std::cerr << "[SEQAN INFO] :: Thank you for using SeqAn!\n"
+                      << "[SEQAN INFO] :: You might want to regsiter you app for support and version check features?!\n"
+                      << "[SEQAN INFO] :: Just send us an email to seqan@team.fu-berlin.de with your app name and version number.\n"
+                      << "[SEQAN INFO] :: If you don't want to recieve this message anymore set --version_check OFF\n\n";
+            myfile.close();
+            return false;
+        }
+        else
+        {
+            myfile.close();
+            return false;
+        }
+    }
+    else
+    {
+        return false;
+    }
+    return true;
+}
+
+// ----------------------------------------------------------------------------
+// Function _isSmaller()
+// ----------------------------------------------------------------------------
+inline bool _isSmaller(seqan::String<int> & left, seqan::String<int> & right)
+{
+    for (unsigned i = 0; i < length(left); ++i)
+    {
+        if (left[i] < right[i])
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+// ----------------------------------------------------------------------------
+// Function _callServer()
+// ----------------------------------------------------------------------------
+inline bool _callServer(VersionCheck me, bool version_file_exists)
+{
+    // system call
+    // http response is stored in a file '.config/seqan/{app_name}_version'
+    if (system(me._command.c_str()))
+    {
+        std::string version_file = me._path + "/" + me._name + "_version.txt";
+        if (!version_file_exists)
+        {
+            std::ofstream version_file_out (version_file.c_str());
+            if (version_file_out.is_open())
+            {
+                version_file_out << "UNABLE TO CALL HOME.\n";
+                version_file_out.close();
+            }
+        }
+        else // just set last modified date to current and do not override possible version information
+        {
```

Comment:
So there was no successful connection to the server, but there is already a file, so you update the timestamp of that file  to call first next day. 
But there is always a file. Then, why not storing the timestamp in the file. We can do this without the OS voodoo but with standard functions of the STL.